tests/baselines/reference/user/puppeteer.log

```diff
@@ -1,96 +1,9 @@
-  Type 'Worker' is missing the following properties from type 'Worker': onmessage, onmessageerror, postMessage, terminate, and 4 more.
-lib/Page.js(142,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(217,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(385,20): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(448,7): error TS2740: Type '(...args: any[]) => Promise<any>' is missing the following properties from type 'Window': applicationCache, clientInformation, closed, customElements, and 221 more.
-lib/Page.js(458,9): error TS2349: This expression is not callable.
-  Type 'Window' has no call signatures.
-lib/Page.js(494,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(504,22): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(517,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(527,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(552,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(605,14): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(941,19): error TS2503: Cannot find namespace 'Protocol'.
-lib/Page.js(1352,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/PipeTransport.js(19,17): error TS1003: Identifier expected.
-lib/Target.js(23,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/Target.js(87,7): error TS2322: Type 'Promise<Worker | Worker>' is not assignable to type 'Promise<Worker>'.
-  Type 'Worker | Worker' is not assignable to type 'Worker'.
-    Type 'Worker' is not assignable to type 'Worker'. Two different types with this name exist, but they are unrelated.
-lib/Target.js(135,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/WebSocketTransport.js(19,17): error TS1003: Identifier expected.
-lib/WebSocketTransport.js(32,72): error TS2345: Argument of type 'import("/puppeteer/puppeteer/node_modules/@types/ws/index")' is not assignable to parameter of type 'WebSocket'.
-  Property 'dispatchEvent' is missing in type 'import("/puppeteer/puppeteer/node_modules/@types/ws/index")' but required in type 'WebSocket'.
-lib/Worker.js(25,50): error TS2503: Cannot find namespace 'Protocol'.
-lib/Worker.js(26,24): error TS2503: Cannot find namespace 'Protocol'.
-lib/Worker.js(33,26): error TS2503: Cannot find namespace 'Protocol'.
-lib/externs.d.ts(18,26): error TS2503: Cannot find namespace 'Protocol'.
-lib/externs.d.ts(18,69): error TS2503: Cannot find namespace 'Protocol'.
-lib/externs.d.ts(19,28): error TS2503: Cannot find namespace 'Protocol'.
-lib/externs.d.ts(19,81): error TS2503: Cannot find namespace 'Protocol'.
-lib/externs.d.ts(19,121): error TS2503: Cannot find namespace 'Protocol'.
-lib/helper.js(45,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/helper.js(63,15): error TS2503: Cannot find namespace 'Protocol'.
-lib/helper.js(89,15): error TS2503: Cannot find namespace 'Protocol'.
+src/Launcher.js(181,17): error TS1003: Identifier expected.
```

Comment:
no, it's just that `@implements` expects an entity name (dotted name), not a arbitrary type reference, unlike `@param` or `@type`.

@Timvdlippe I'd prefer not to change this if it's not going to cause a problem for you, since you're the first customer to use `@implements`. I guess that it's not a problem since there's only one instance in over 300 `@implements`, even on this ancient snapshot of the code.

Reply:
Yes that's fair. And also this is Puppeteer, which is moving to TypeScript in the coming week CC @jackfranklin

Reply:
Oops, I got confused between puppeteer and chrome-devtools-frontend. Even in the ancient clone of the latter, there are no instances of this error.

Our clone of puppeteer is on master, so this is an error that will show up in 3.9.